src/node/filesystem/binary.node.ts

```diff
@@ -0,0 +1,341 @@
+export class BinaryData
+  implements
+    IBinaryData,
+    IBinaryFileMeta,
+    IDestructible,
+    IInitializable
+{
+  private buffer: Buffer;
+   
+  public static fromPath(filename: string, options: { filename?: string, mimetype?: string } = {}): BinaryData {
+    return new BinaryData(new FileContainer(filename, options));
+  }
+
+  public static fromStream(stream: Readable): BinaryData {
+    return new BinaryData(new StreamContainer(stream));
+  }
+
+  private constructor(private dataContainer: IDataContainer) {
+    this.buffer = Buffer.from([]);
+  }
+
+  public get filename(): string | undefined {
+    if (isBinaryFileMeta(this.dataContainer)) {
+      return this.dataContainer.filename;
+    }
+
+    return undefined;
+  }
+
+  public get mimetype(): string | undefined {
+    if (isBinaryFileMeta(this.dataContainer)) {
+      return this.dataContainer.mimetype;
+    }
+
+    return undefined;
+  }
+
+  public get filesize(): number | undefined {
+    if (isBinaryFileMeta(this.dataContainer)) {
+      return this.dataContainer.filesize;
+    }
+
+    return undefined;
+  }
+
+  public async initialize(): Promise<void> {
+    if (isInitializable(this.dataContainer)) {
+      await this.dataContainer.initialize();
+    }
+  }
+
+  public async destroy(): Promise<void> {
+    if (isDestructible(this.dataContainer)) {
+      await this.dataContainer.destroy();
+    }
+
+    this.buffer = Buffer.from([]);
+  }
+
+  private async fillBuffer(sizeAtLeast: number): Promise<void> {
+    if (this.buffer.length < sizeAtLeast) {
+      const read = await this.dataContainer.read(sizeAtLeast - this.buffer.length);
+      if (read !== undefined) {
+        this.buffer = Buffer.concat([this.buffer, read]);
+      }
+    }
+  }
+
+  private consumeBuffer(size?: number): Buffer {
+    let data: Buffer;
+
+    if (size === undefined) {
+      data = this.buffer;
+      this.buffer = Buffer.from([]);
+    } else {
+      data = this.buffer.subarray(0, size);
+      this.buffer = this.buffer.subarray(size);
+    }
+
+    return data;
+  }
+
+  public chunkBy(chunkSize: number): AsyncIterable<Buffer> {
+    if (
+      chunkSize === undefined ||
+      chunkSize === null ||
+      typeof chunkSize !== 'number' ||
+      chunkSize <= 0 ||
+      isNaN(chunkSize)
+    ) {
+      throw new UnexpectedError('Invalid chunk size');
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+
+    return {
+      [Symbol.asyncIterator]: () => ({
+        async next() {          
+          await self.fillBuffer(chunkSize);
+          const data = self.consumeBuffer(chunkSize);
+
+          if (!data.length) {
+            return { done: true, value: undefined };
+          }
+
+          return { done: false, value: data };
+        },
+      }),
+    };
+  }
+
+  public async getAllData(): Promise<Buffer> {
+    let size = this.buffer.length;
+
+    while (this.buffer.length >= size) {
+      size += 1000; // TODO constant
```

Comment:
thoughts: ideally a configurable constant, and probably should be bigger like 16 or even 64kB to make sense for bigger files. Although we might have bottlenecks in other places anyway.

Reply:
You are right. I left this for now to hand over rest of changes. But thanks for suggestions.

Reply:
Yes, you are right. I left it as to-do, because wanted to hand over the rest.

Reply:
Added chunkSize argument with 16000 (16kB) as default